Allow CLI invocation of Cron

pradtke opened the pull request:
Make it easier to run SSP cron functionality from CLI.
This allows setting memory_limit and max_execution_time values that are appropriate for cron functionality that you don't wanted enabled for the all requests.

For tests, I'm currently blocked on getting module enabled during testing. See discussion https://groups.google.com/forum/#!topic/simplesamlphp-dev/hqh7VlSSzhw

thijskh commented on modules/cron/bin/cron.php:
```diff
@@ -0,0 +1,45 @@
+}
+
+$options = getopt("t:");
+if (posix_getuid() === 0) {
```
Probably add an `function_exists` there because `posix_getuid` is not always available (requires posix module, not on Windows).

pradtke replied:
thanks for the tip. I've added the check.

pradtke commented:
@thijskh I've updated the documentation with example invocation and with some advice/warnings on file permissions and logging.

I've also incorporated the function_exists() check.

thijskh commented on modules/cron/docs/cron.md:
```diff
@@ -80,3 +88,35 @@ follow the appropriate links to execute the cron jobs you want. The
+
+Below is an example of invoking the script. It will:
+
+* Run a command as the apache user
```
Might be a nit, but I'd write this as `apache` to make it abundantly clear that this is actually the user named 'apache'.

pradtke replied:
Make sense. I've added the `code` tags.

jaimeperez commented on modules/cron/lib/Cron.php:
```diff
@@ -0,0 +1,61 @@
+/**
+ * Handles interactions with SSP's cron system/hooks.
+ */
+class sspmod_cron_Cron
```
Since this is a new class (and the only one in this module), I would actually prefer it to use the PSR-4-compliant naming convention. That would mean naming the class `Cron` inside the `SimpleSAML\Module\cron` namespace.

jaimeperez commented:
I've added a minor comment on how to name the new Cron class. Sorry for the late reaction from my side!

pradtke commented:
@jaimeperez no worries. I added in the namespace.

jaimeperez commented:
Merged! Thanks a lot for your work Patrick, and sorry so much for the delay on my side.
